packages/shell-api/src/collection.ts

```diff
@@ -2033,4 +2034,29 @@ export default class Collection extends ShellApiWithMongoClass {
     this._emitCollectionApiCall('unhideIndex');
     return setHideIndex(this, index, false);
   }
+
+  @returnsPromise
+  @topologies([Topologies.ReplSet, Topologies.Sharded])
+  @apiVersions([])
+  async analyzeShardKey(key: Document): Promise<Document> {
+    assertArgsDefinedType([key], [true], 'Collection.analyzeShardKey');
+    this._emitCollectionApiCall('analyzeShardKey', { key });
+    return await this._database._runAdminCommand({
+      analyzeShardKey: this.getFullName(),
+      key
+    });
+  }
+
+  @returnsPromise
+  @topologies([Topologies.ReplSet, Topologies.Sharded])
+  @apiVersions([])
+  async configureQueryAnalyzer({ mode, sampleRate }: { mode: 'full' | 'off', sampleRate?: number }): Promise<Document> {
+    assertArgsDefinedType([mode, sampleRate], ['string', [undefined, 'number']], 'Collection.configureQueryAnalyzer');
+    this._emitCollectionApiCall('configureQueryAnalyzer', { mode, sampleRate });
+    return await this._database._runAdminCommand({
+      configureQueryAnalyzer: this.getFullName(),
+      mode,
+      sampleRate: mode === 'full' && sampleRate ? new bson.Double(sampleRate) : undefined
+    });
```

Comment:
Is there a specific reason for now passing through the entire options object, i.e. any extra keys that we don’t know about? That might be nice for forward compatibility with options we don’t know about yet.